src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -26,18 +29,27 @@ public class ConsoleLoggerProvider : ILoggerProvider, ISupportExternalScope
         /// </summary>
         /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
         public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
+            : this(options, Enumerable.Empty<ConsoleFormatter>()) { }
+
+        /// <summary>
+        /// Creates an instance of <see cref="ConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="ConsoleLogger"/> instances with.</param>
+        /// <param name="formatters">Log formatters added for <see cref="ConsoleLogger"/> insteaces.</param>
+        public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IEnumerable<ConsoleFormatter> formatters)
         {
             _options = options;
             _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
+            SetFormatters(formatters);
 
             ReloadLoggerOptions(options.CurrentValue);
             _optionsReloadToken = _options.OnChange(ReloadLoggerOptions);
 
             _messageQueue = new ConsoleLoggerProcessor();
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!DoesConsoleSupportAnsi())
```

Comment:
It might make sense to flip this if check around:

```C#
if (DoesSupportAnsi())
{ 
   // use ansi
}
else
{
 // parse ansi
}
```

reads easier.